packages/swapper/src/utils.ts

```diff
@@ -169,16 +170,25 @@ export const checkEvmSwapStatus = async ({
   txHash,
   chainId,
   assertGetEvmChainAdapter,
+  getSafeApiKit,
 }: {
   txHash: string
   chainId: ChainId
   assertGetEvmChainAdapter: (chainId: ChainId) => EvmChainAdapter
+  getSafeApiKit: (chainId: EvmChainId, safeAddress: string) => Promise<SafeApiKit>
 }): Promise<{
   status: TxStatus
   buyTxHash: string | undefined
   message: string | undefined
 }> => {
   try {
+    const safeApiKit = await getSafeApiKit(
+      chainId as EvmChainId,
+      '0x00342Dc97c6419089130bB2F1a39c1Bfea4fFe40',
+    )
+    // @ts-ignore
+    const transaction = await safeApiKit.getTransaction(txHash)
```

Comment:
Note, alternatively we could use https://safe-transaction-avalanche.safe.global/api/v1/multisig-transactions/0xa75d5c9b8fd53cc74a0e35fba841ca1c6f5b8f2ee665a0ac8cf8f93e631a1200/ and its variant on other chains to avoid a few JSON-RPC calls and first instantiations which may take a few seconds.

Also avoids the plumbing while at it!